### ISSUE TITLE
Haskell syntax coloring bug

### ISSUE THREAD

**Issue by machinedgod** [martanne/vis#888]
Hi,
there's a small syntax coloring bug in haskell colorizer, specifically regarding the `--` operator which marks beginning of the line comment.
There are various other operators (practically an infinite number, since they can be user-created) that start with `--`, that aren't comments, for example `--|` for documentation, or `-->` as an operator in xmonad.

I am not quite sure if a trailing space is required to specifically start a comment, but perhaps, this would be a nice quick fix since I've never seen a comment in the wild without one.

So to sum it up:
* `--anything`  not a comment
* `-- anything` comment

**Comment by erf** [comment]
You could change that in lua/lexers/haskell.lua

Using Lua pattern, try to change "--" to "--%s"

**Comment by machinedgod** [comment]
Hi @erf,
thank you for the tip, I'll submit the patch as well.
